tests/baselines/reference/assignmentToObjectAndFunction.errors.txt

```diff
@@ -14,6 +16,7 @@ tests/cases/compiler/assignmentToObjectAndFunction.ts(29,5): error TS2322: Type
 !!! error TS2322: Type '{ toString: number; }' is not assignable to type 'Object'.
 !!! error TS2322:   Types of property 'toString' are incompatible.
 !!! error TS2322:     Type 'number' is not assignable to type '() => string'.
+!!! error TS2322:       Type 'Number' provides no match for the signature '(): string'

```

Comment:
Some of these become a little excessive. Do we really need to elaborate when one of the types is a primitive? The only case that would potentially backfire is if you had an alias to a primitive type, in which case I think you actually would want to elaborate.